src/sentry/relay/config/metric_extraction.py

```diff
@@ -145,31 +184,93 @@ def _convert_alert_to_metric(alert: AlertRule) -> Optional[MetricSpec]:
     if not alert.is_custom_metric:
         return None
 
-    # TODO: Determine these based on the alert.snuba_query.field:
-    metric_type = "c"
-    metric_unit = "none"
-    field = None
+    try:
+        query = alert.snuba_query.query
+        field, metric_type = _extract_field_info(alert.snuba_query.aggregate)
+
+        condition = _convert_alert_query_to_condition(query)
+        query_hash = _get_query_hash(field, query)
+
+        return {
+            "category": DataCategory.TRANSACTION.api_name(),
+            "mri": f"{metric_type}:{CUSTOM_ALERT_METRIC_NAME}@none",
+            "field": field,
+            "condition": condition,
+            "tags": [{"key": "query_hash", "value": query_hash}],
+        }
+    except Exception as e:
+        logger.error(e, exc_info=True)
+        return None
+
 
-    condition = _convert_alert_query_to_condition(alert.snuba_query.query)
-    query_hash = str(alert.id)  # TODO: Hash field + condition
+def _extract_field_info(aggregate: str) -> Tuple[Optional[str], str]:
+    select = _get_query_builder().resolve_column(aggregate, False)
 
-    return {
-        "category": DataCategory.TRANSACTION.api_name(),
-        "mri": f"{metric_type}:{CUSTOM_ALERT_METRIC_NAME}@{metric_unit}",
-        "field": field,
-        "condition": condition,
-        "tags": [{"key": "query_hash", "value": query_hash}],
-    }
+    metric_type = _AGGREGATE_TO_METRIC_TYPE.get(select.function)
+    assert metric_type, f"Unsupported aggregate function {select.function}"
+
+    if metric_type == "c":
+        assert not select.parameters, "Count should not have parameters"
+        return None, metric_type
+    else:
+        assert len(select.parameters) == 1, "Only one parameter is supported"
+
+        name = select.parameters[0].name
+        assert name in _SNUBA_TO_RELAY_FIELDS, f"Unsupported field {name}"
+
+        return _SNUBA_TO_RELAY_FIELDS[name], metric_type
 
 
 def _convert_alert_query_to_condition(query: str) -> RuleCondition:
-    # TODO: Parse and convert the query using QueryBuilder
+    where, having = _get_query_builder().resolve_conditions(query, False)
```

Comment:
Should we fail early if the query filters on a different event type from transactions?